### PR TITLE
feat(coinjoin): group mixing transactions on home screen

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

Group CoinJoin-related transactions to one group per day.
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/java/org/dash/wallet/common/transactions/TransactionWrapperFactory.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.dash.wallet.common.transactions
+
+import org.bitcoinj.core.Transaction
+
+interface TransactionWrapperFactory {
+    fun tryInclude(tx: Transaction): Pair<Boolean, TransactionWrapper?>
+}
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/transactions/TransactionWrapperFactory.kt:24]: This wrapper factory will determine which wrapper should be used and return it along with the results of `tryInclude` from the wrapper.

File: wallet/src/de/schildbach/wallet/transactions/TransactionWrapperHelper.kt [hunk 2]
```diff
@@ -20,15 +20,19 @@ package de.schildbach.wallet.transactions
...
     fun wrapTransactions(
         transactions: Set<Transaction?>,
-        vararg wrappers: TransactionWrapper
+        vararg wrapperFactories: TransactionWrapperFactory
     ): Collection<TransactionWrapper> {
         val wrappedTransactions = ArrayList<TransactionWrapper>()
-
+        log.info("processing transaction wrappers")
         for (transaction in transactions) {
+            // log.info("wrapper ------------------ \n  {}", transaction?.txId)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/transactions/TransactionWrapperHelper.kt:35]: This logging will be removed before the PR is finalized.

File: wallet/src/de/schildbach/wallet/transactions/coinjoin/CoinJoinMixingFilter.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                when {
+                    ScriptPattern.isP2PKH(output.scriptPubKey) ->
+                        wallet.coinJoin.findKeyFromPubKeyHash(
+                            ScriptPattern.extractHashFromP2PKH(output.scriptPubKey),
+                            Script.ScriptType.P2PKH
+                        ) != null
+                    ScriptPattern.isOpReturn(output.scriptPubKey) -> true
+                    else -> false
+                }
+            }*/
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/transactions/coinjoin/CoinJoinMixingFilter.kt:40]: We don't need to check the keys.

File: wallet/src/de/schildbach/wallet/ui/transactions/TransactionGroupDetailsFragment.kt [hunk 4]
```diff
@@ -52,11 +53,19 @@ class TransactionGroupDetailsFragment() : OffsetDialogFragment(R.layout.transact
...
+                binding.icon.setImageResource(R.drawable.ic_crowdnode_logo)
+                binding.detailsTitle.text = getString(R.string.crowdnode_tx_set_title)
+                binding.detailsMessage.text = getString(R.string.crowdnode_tx_set_explainer)
+            }
+            is CoinJoinMixingTxSet -> {
+                binding.groupTitle.text = getString(R.string.coinjoin_mixing_transactions)
+                binding.icon.setImageResource(R.drawable.ic_coinjoin)
+                binding.detailsTitle.text = getString(R.string.crowdnode_tx_set_title)
+                binding.detailsMessage.text = getString(R.string.coinjoin_transaction_group)
+            }
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/transactions/TransactionGroupDetailsFragment.kt:68]: At some point we should move it to TxResourceMapper too to keep to open-closed principle

**Review by Syn-McJ — APPROVED** [review]
Looks good

**Pull request merged**